### ISSUE TITLE
x11.fcasts unable to be set to 0 or 1

### ISSUE THREAD

**Issue by pmbyron** [jdemetra/rjdemetra#42]
When using the x13_spec_def function, it will happily take a value of 0 or 1 for x11.fcasts and create the specification, but then when I use that specification in the x13 function, I get the following errors:

If x11.fcasts = 0:
Error in attr(x, "tsp") <- c(1, NROW(x), 1) : 
  attempt to set an attribute on NULL

If x11.fcasts = 1:
Error in `colnames<-`(`*tmp*`, value = c(finals_ts_names2)) : 
  attempt to set 'colnames' on an object with less than two dimensions

I couldn't see any other method of turning off forecasting, as "X11" isn't available as one of the default specifications (there's only "RSA0" through to "RSA5c").

(By the way, thanks for building this package, it's wonderful to be able to run JD+ without having to deal directly with Java objects.)

**Comment by AQLT** [comment]
Indeed there is an error in the code. I've corrected it in my personal repository: https://github.com/AQLT/rjdemetra. I will soon do a pull request soon with some other changes.

**Comment by pmbyron** [comment]
Hi Alain, Thank you very much, particularly for being so prompt.

Peter.

**Comment by AQLT** [comment]
Hi Peter, it's corrected in the last pull request (#43). Note that since this pull request all the _def functions are now deprecated: use `x13_spec` instead of `x13_spec_def`
``` r
library(RJDemetra)
myspec1 <- x13_spec(x11.fcasts = 0)
x13_mod1 <- x13(ipi_c_eu[,1], myspec1)
x13_mod1$final
#> Last observed values
#>              y       sa        t           s           i
#> Jan 2017  96.9 103.6767 103.3854  -6.7766920  0.29128813
#> Feb 2017  99.4 102.7751 103.4763  -3.3751319 -0.70115772
#> Mar 2017 111.2 103.7306 103.6384   7.4694449  0.09215246
#> Apr 2017 103.7 103.6293 103.9118   0.0707067 -0.28255595
#> May 2017 105.5 104.7018 104.3171   0.7981688  0.38469016
#> Jun 2017 108.4 103.6319 104.8330   4.7681143 -1.20114781
#> Jul 2017 105.9 105.1936 105.4306   0.7064323 -0.23702558
#> Aug 2017  91.7 105.9746 106.0608 -14.2746297 -0.08614573
#> Sep 2017 112.2 106.6653 106.6952   5.5347230 -0.02991125
#> Oct 2017 113.5 107.6234 107.3039   5.8766022  0.31950978
#> Nov 2017 114.2 109.3897 107.8407   4.8102510  1.54900689
#> Dec 2017 102.7 108.3771 108.2967  -5.6770936  0.08034628
#> 
#> No forecast
myspec2 <- x13_spec(spec=myspec1, x11.fcasts = 1)
x13_mod2 <- x13(ipi_c_eu[,1], myspec2)
x13_mod2$final
#> Last observed values
#>              y       sa        t            s           i
#> Jan 2017  96.9 103.3477 103.3211  -6.44768199  0.02659887
#> Feb 2017  99.4 102.7711 103.4205  -3.37110703 -0.64938322
#> Mar 2017 111.2 103.7982 103.6074   7.40178575  0.19082016
#> Apr 2017 103.7 103.6313 103.9130   0.06867516 -0.28171358
#> May 2017 105.5 104.7206 104.3447   0.77943544  0.37589040
#> Jun 2017 108.4 103.6514 104.8746   4.74859733 -1.22322494
#> Jul 2017 105.9 105.2389 105.4542   0.66112554 -0.21537429
#> Aug 2017  91.7 106.0043 106.0750 -14.30434304 -0.07064623
#> Sep 2017 112.2 106.6896 106.7353   5.51044556 -0.04576724
#> Oct 2017 113.5 107.6691 107.4139   5.83085293  0.25523373
#> Nov 2017 114.2 109.4097 108.0742   4.79034687  1.33544589
#> Dec 2017 102.7 108.4754 108.6783  -5.77540176 -0.20293726
#> 
#> Forecasts:
#>               y_f     sa_f      t_f       s_f       i_f
#> Jan 2018 103.6428 109.6334 109.1808 -5.990599 0.4526157
```

<sup>Created on 2019-03-25 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>
